### PR TITLE
Builder for Task

### PR DESCRIPTION
**Pull request by matthiasbeyer**
@colin-kiegel can you help me, I'm not sure what the error is here.

### CODE REVIEW

File: src/task.rs [hunk 1]
```diff
@@ -262,6 +262,64 @@ impl Task {
...
+            tags        : Option<Vec<Tag>>,
+            until       : Option<Date>,
+            wait        : Option<Date>,
+        }
+    }
+
+    impl TryInto<Task> for TaskBuilder {
+        type Error = TaskBuilderError;
+
+        fn try_into(self) -> Result<Task, TaskBuilderError> {
```
Comment (colin-kiegel) [inline src/task.rs]: `v0.3.0` implements a `build(self) -> Result<Task, String>` method on `TaskBuilder` for you. You don't need to implement this anymore, unless you don't like something about the canonical implemention from `derive_builder`

File: src/task.rs [hunk 2]
```diff
@@ -424,5 +482,20 @@ r#"{
...
 
+    #[test]
+    fn test_builder_simple() {
+        use task::builder::TaskBuilder;
+
+        let t = TaskBuilder::default()
+            .with_status(TaskStatus::Pending)
+            .with_description("test".into_string())
+            .with_entry(mkdate("20150619T165438Z"))
+            .try_into().unwrap();
```
Comment (colin-kiegel) [inline src/task.rs]: just replace `try_into()` with `build()`.

But be aware that the current implementation assumes that all fields are mandatory. I.e. this test will yield an error about not all fields being initialized. This is the intended behaviour right now.

We will likely introduce an option to make fields voluntary and provide default values or fallback to the `Default` trait.

**Comment by colin-kiegel** [comment]
PS: In a nutshell you were mostly using the API of `v0.2.1` which is superseded in `v0.3.0` because we switched from macros by example to macros 1.1.

You can have a look at the [updated documentation](https://docs.rs/derive_builder/0.3.0/derive_builder/) or [examples/tests](https://github.com/colin-kiegel/rust-derive-builder/tree/v0.3.0/derive-builder-test/tests). :-)

**Comment by matthiasbeyer** [comment]
Ah, awesome! Thanks a lot for your comments!

**Comment by colin-kiegel** [comment]
Ok, cool. Let me know if you require additional features for your use case.

**Comment by matthiasbeyer** [comment]
It seems like serde and the builder do not work well together :rofl:

**Comment by colin-kiegel** [comment]
It looks like [this](https://travis-ci.org/matthiasbeyer/task-hookrs/jobs/201054844#L424-L427) is your only problem.

Workaround: `#![allow(missing_docs)]` in `/src/task.rs`. I'll look into improving derive_builder [here](https://github.com/colin-kiegel/rust-derive-builder/issues/37).

**Comment by maralorn** [comment]
Obsolete because of #65 .

**Pull request closed without merging**